src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/MemberAndDeclarationInfo.cs

```diff
@@ -78,6 +78,53 @@ public MemberAndDeclarationInfo(MemberInfo member, DkmClrDebuggerBrowsableAttrib
             _inheritanceLevel = inheritanceLevel;
         }
 
+        // Obtains the set of the required and optional custom modifiers concerning
+        // this member.  This is required by managed C++, as the compiler emits
+        // modopts for special types, such as C++ references (e.g. 'int&') and 
+        // boxed value-types (e.g. 'int^').  In the core result provider,
+        // we simply treat the list of modopts and modreqs as opaque values
+        // and plumb them on through via the DkmClrCustomTypeInfo.  The language-
+        // specific portion of the EE is responsible for interpereting these
+        // custom modifiers and inferring language-specific semantics from them.
+        public void GetCustomModifiers(out Type[] modopts, out Type[] modreqs)
+        {
+            switch(MemberType)
+            {
+                case MemberTypes.Property:
+                    PropertyInfo pi = (PropertyInfo)_member;
+                    if (pi != null)
+                    {
+                        var getMethod = pi.GetGetMethod();
+                        if (getMethod != null)
+                        {
+                            var returnParam = getMethod.ReturnParameter;
+                            if (returnParam != null)
+                            {
+                                modopts = returnParam.GetOptionalCustomModifiers();
+                                modreqs = returnParam.GetRequiredCustomModifiers();
+                                return;
+                            }
+                        }
+                    }
+                    break;
+                case MemberTypes.Field:
+                    FieldInfo fi = (FieldInfo)_member;
+                    if (fi != null)
```

Comment:
Isn't this null check unneeded?  The field is readonly and checked for null in the ctor